## PR #941 Re-Review: BLOCKING

The fetch-before-mutate fix in `SentScheduledItemAsync` is architecturally correct, but it broke two existing unit tests that were not updated to match the new behaviour. The CI build-and-test job fails.

### Blocking: Two failing unit tests

`ScheduledItemManagerTests.cs` has two tests written for the old (no-fetch) implementation:

**Test 1** — `SentScheduledItemAsync_WithIdOnly_ShouldCallRepositoryWithUtcNow`:

```csharp
_repository.Setup(r => r.SentScheduledItemAsync(1, It.IsAny<DateTimeOffset>())).ReturnsAsync(true);
// GetAsync is NOT mocked → Moq returns null → manager returns false early
var result = await _scheduledItemManager.SentScheduledItemAsync(1);
Assert.True(result);  // FAILS: result is false
```

**Test 2** — `SentScheduledItemAsync_WithIdAndDate_ShouldCallRepository`:

```csharp
var sentOn = DateTimeOffset.UtcNow;
_repository.Setup(r => r.SentScheduledItemAsync(1, sentOn)).ReturnsAsync(true);
// GetAsync is NOT mocked → Moq returns null → manager returns false early
var result = await _scheduledItemManager.SentScheduledItemAsync(1, sentOn);
Assert.True(result);  // FAILS: result is false
```

**Fix — add this `Arrange` line to both tests:**

```csharp
var entity = new ScheduledItem { Id = 1 };
_repository.Setup(r => r.GetAsync(1, default)).ReturnsAsync(entity);
```

---

### Important: CS8613 compiler warning

```
warning CS8613: Nullability of reference types in return type of
'Task<ScheduledItem?> ScheduledItemManager.GetAsync(...)'
doesn't match implicitly implemented member
'Task<ScheduledItem> IManager<ScheduledItem>.GetAsync(...)'.
```

`ScheduledItemManager.GetAsync` returns `Task<ScheduledItem?>` (nullable) but `IManager<T>` declares `Task<T>` (non-nullable). Fix by either:

- Reverting `ScheduledItemManager.GetAsync` to return `Task<ScheduledItem>`, or
- Updating `IManager<T>` to `Task<T?>` and propagating to all implementors.

---

### Previously-blocking issues confirmed fixed

- `SentScheduledItemAsync` fetch-before-mutate logic is correct
- `InvalidateUserCaches` is called only on the successful path
- All test constructors use `new MemoryCache(new MemoryCacheOptions())` — not `Mock<IMemoryCache>`
- No hardcoded secrets, no CSRF issues

> Reviewed by Neo

--- a/src/JosephGuadagno.Broadcasting.Managers.Tests/ScheduledItemManagerTests.cs
+++ b/src/JosephGuadagno.Broadcasting.Managers.Tests/ScheduledItemManagerTests.cs
@@ -206,6 +206,8 @@ public class ScheduledItemManagerTests
     public async Task SentScheduledItemAsync_WithIdOnly_ShouldCallRepositoryWithUtcNow()
     {
         // Arrange
+        var entity = new ScheduledItem { Id = 1 };
+        _repository.Setup(r => r.GetAsync(1, default)).ReturnsAsync(entity);
         _repository.Setup(r => r.SentScheduledItemAsync(1, It.IsAny<DateTimeOffset>())).ReturnsAsync(true);
 
         // Act
@@ -220,6 +222,8 @@ public class ScheduledItemManagerTests
     public async Task SentScheduledItemAsync_WithIdAndDate_ShouldCallRepository()
     {
         // Arrange
+        var entity = new ScheduledItem { Id = 1 };
+        _repository.Setup(r => r.GetAsync(1, default)).ReturnsAsync(entity);
         var sentOn = DateTimeOffset.UtcNow;
         _repository.Setup(r => r.SentScheduledItemAsync(1, sentOn)).ReturnsAsync(true);
 

--- a/src/JosephGuadagno.Broadcasting.Managers/ScheduledItemManager.cs
+++ b/src/JosephGuadagno.Broadcasting.Managers/ScheduledItemManager.cs
@@ -26,7 +26,7 @@ public class ScheduledItemManager : IScheduledItemManager
         _cache = cache;
     }
 
-    public async Task<ScheduledItem?> GetAsync(int primaryKey, CancellationToken cancellationToken = default)
+    public async Task<ScheduledItem> GetAsync(int primaryKey, CancellationToken cancellationToken = default)
     {
         return await _scheduledItemDataStore.GetAsync(primaryKey, cancellationToken);
     }

